packages/turf-boolean-contains/index.js

```diff
@@ -0,0 +1,316 @@
+        case 'MultiPoint':
+            return isMultiPointOnLine(geom1, geom2);
+        }
+        throw new Error('feature2 ' + type2 + ' geometry not supported');
+    case 'Polygon':
+        switch (type2) {
+        case 'Point':
+            return inside(geom2, geom1, true);
+        case 'LineString':
+            return isLineInPoly(geom1, geom2);
+        case 'Polygon':
+            return isPolyInPoly(feature2, feature1);
+        case 'MultiPoint':
+            return isMultiPointInPoly(geom1, geom2);
+        }
+        throw new Error('feature2 ' + type2 + ' geometry not supported');
+    default:
+        throw new Error('feature1 ' + type1 + ' geometry not supported');
+    }
+};
+
+function isPointInMultiPoint(multiPoint, point) {
+    var i;
+    var output = false;
+    for (i = 0; i < multiPoint.coordinates.length; i++) {
+        if (compareCoords(multiPoint.coordinates[i], point.coordinates)) {
+            output = true;
+            break;
+        }
+    }
+    return output;
+}
+
+function isMultiPointInMultiPoint(multiPoint1, multiPoint2) {
+    var foundAMatch = 0;
+    for (var i = 0; i < multiPoint2.coordinates.length; i++) {
+        var anyMatch = false;
+        for (var i2 = 0; i2 < multiPoint1.coordinates.length; i2++) {
+            if (compareCoords(multiPoint2.coordinates[i], multiPoint1.coordinates[i2])) {
+                foundAMatch++;
+                anyMatch = true;
+                break;
+            }
+        }
+        if (!anyMatch) {
+            return false;
+        }
+    }
+    return foundAMatch > 0;
+}
+
+// http://stackoverflow.com/a/11908158/1979085
+function isPointOnLine(lineString, point) {
+    var output = false;
+    for (var i = 0; i < lineString.coordinates.length - 1; i++) {
+        var incEndVertices = true;
+        if (i === 0 || i === lineString.coordinates.length - 2) {
+            incEndVertices = false;
+        }
+        if (isPointOnLineSegment(lineString.coordinates[i], lineString.coordinates[i + 1], point.coordinates, incEndVertices)) {
+            output = true;
+            break;
+        }
+    }
+    return output;
+}
+
+function isMultiPointOnLine(lineString, multiPoint) {
+    var output = true;
+    for (var i = 0; i < multiPoint.coordinates.length; i++) {
+        var pointIsOnLine = false;
+        for (var i2 = 0; i2 < lineString.coordinates.length - 1; i2++) {
+            var incEndVertices = true;
+            if (i2 === 0 || i2 === lineString.coordinates.length - 2) {
+                incEndVertices = false;
+            }
+            if (isPointOnLineSegment(lineString.coordinates[i2], lineString.coordinates[i2 + 1], multiPoint.coordinates[i], incEndVertices)) {
+                pointIsOnLine = true;
+                break;
+            }
+        }
+        if (!pointIsOnLine) {
+            output = false;
+            break;
+        }
+    }
+    return output;
+}
+
+function isMultiPointInPoly(polygon, multiPoint) {
```

Comment:
@rowanwins Is the goal to implement the DE-9IM definitions of these functions? Because this approach is not implementing `contains` definition. There is no check that any point falls in the interior. This allows all points to fall on boundary.